dalm/training/generator_only/trainer.py

```diff
@@ -0,0 +1,299 @@
+import argparse
+import logging
+import os
+from typing import Any, Callable, Dict, Optional, Tuple
+
+import torch
+from accelerate import Accelerator
+from datasets import Dataset, load_dataset, load_from_disk
+from peft import LoraConfig
+from tqdm import tqdm
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    BitsAndBytesConfig,
+    PreTrainedTokenizerBase,
+    TrainingArguments,
+)
+from trl import SFTTrainer  # type: ignore[import]
+
```

Comment:
Can you add `import wandb` here?  This will fail fast when `wandb` is not installed, rather than failing in the middle of a pipeline with this stacktrace:

```
  File "/opt/conda/lib/python3.10/site-packages/dalm/pipelines/reading_comprehension_pipeline.py", line 180, in pipeline
    train_generator(
  File "/opt/conda/lib/python3.10/site-packages/dalm/training/generator_only/trainer.py", line 240, in train_generator
    trainer = SFTTrainer(
  File "/opt/conda/lib/python3.10/site-packages/trl/trainer/sft_trainer.py", line 252, in __init__
    super().__init__(
  File "/opt/conda/lib/python3.10/site-packages/transformers/trainer.py", line 525, in __init__
    self.callback_handler = CallbackHandler(
  File "/opt/conda/lib/python3.10/site-packages/transformers/trainer_callback.py", line 306, in __init__
    self.add_callback(cb)
  File "/opt/conda/lib/python3.10/site-packages/transformers/trainer_callback.py", line 323, in add_callback
    cb = callback() if isinstance(callback, type) else callback
  File "/opt/conda/lib/python3.10/site-packages/transformers/integrations/integration_utils.py", line 669, in __init__
    raise RuntimeError("WandbCallback requires wandb to be installed. Run `pip install wandb`.")
RuntimeError: WandbCallback requires wandb to be installed. Run `pip install wandb`.
```

Reply:
I think given the number of options that exists for tracker, separating ourselves from a certain tracker may be the best way forward

Reply:
Yeah, can we just default to no tracking?  Then users can add it as needed, including libs